Changed PostgreSql New-AzPostgreSqlFlexibleServer cmdlet and Added Get-AzPostgreSqlFlexibleServerLocationBasedCapability

DaeunYim opened the pull request:
<!-- DO NOT DELETE THIS TEMPLATE -->

## Description

<!-- Please add a brief description of the changes made in this PR -->

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

msJinLei commented:
@DaeunYim Could you merge https://github.com/Azure/azure-powershell/pull/13552 into this one? I will review them together.

DaeunYim commented:
> @DaeunYim Could you merge #13552 into this one? I will review them together.

They are already integrated here! And I also resolved the same problems as MySql you would want me to fix here!

DaeunYim commented:
> @DaeunYim Could you merge #13552 into this one? I will review them together.

They are already integrated here! And I also resolved the same problems as MySql you would want me to fix here!

DaeunYim commented:
Az.PostgreSql 
Updated New-AzPostgreSqlFlexibleServer cmdlet to provide easy resource group and network management within RDBMS flexible server management. 
Added new cmdlets Get-AzPostgreSqlFlexibleServerLocationBasedCapability and Get-AzPostgreSqlConnectionString

msJinLei commented on src/PostgreSql/custom/Restore-AzPostgreSqlFlexibleServer_PointInTimeRestore.ps1:
```diff
@@ -0,0 +1,130 @@
+        [Parameter(Mandatory, HelpMessage = 'The point in time to restore from (ISO8601 format), e.g., 2017-04-26T02:10:00+08:00.')]
+        [Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Category('Body')]
+        [System.DateTime]
+        ${RestorePointInTime},
```
https://github.com/Azure/azure-powershell/blob/86c11f6258a42ecd0440939ca256ee86e214e576/src/MySql/custom/Restore-AzMySqlFlexibleServer_PointInTimeRestore.ps1#L46

DaeunYim replied:
Any concern here??

msJinLei commented on src/PostgreSql/custom/Update-AzPostgreSqlFlexibleServer.ps1:
```diff
@@ -0,0 +1,202 @@
+            if ($PSBoundParameters.ContainsKey('AdministratorLoginPassword')) {
+                $bStr = . "$PSScriptRoot/../utils/Unprotect-SecureString.ps1" $PSBoundParameters['AdministratorLoginPassword']
+                $null = $PSBoundParameters.Remove('AdministratorLoginPassword')
+                $PSBoundParameters.Add('AdministratorLoginPassword', [System.Runtime.InteropServices.marshal]::PtrToStringAuto($bStr))
```
no need to call PtrToStringAuto

DaeunYim replied:
Removed, thanks.

msJinLei commented on src/PostgreSql/test/Get-AzPostgreSqlConfiguration.Tests.ps1:
```diff
@@ -23,7 +23,7 @@ Describe 'Get-AzPostgreSqlConfiguration' {
     }
 
     It 'GetViaIdentity' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/configurations/datestyle"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Get-AzPostgreSqlFirewallRule.Tests.ps1:
```diff
@@ -29,7 +29,7 @@ Describe 'Get-AzPostgreSqlFirewallRule' {
 
     It 'GetViaIdentity' {
         New-AzPostgreSqlFirewallRule -Name $env.firewallRuleName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -EndIPAddress 0.0.0.1 -StartIPAddress 0.0.0.0
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/firewallRules/$($env.firewallRuleName)"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Get-AzPostgreSqlServer.Tests.ps1:
```diff
@@ -28,7 +28,7 @@ Describe 'Get-AzPostgreSqlServer' {
     }
 
     It 'GetViaIdentity' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Get-AzPostgreSqlVirtualNetworkRule.Tests.ps1:
```diff
@@ -13,25 +13,25 @@ while(-not $mockingPath) {
 
 Describe 'Get-AzPostgreSqlVirtualNetworkRule' {
     It 'List' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/MySqlVNet/subnets/MysqlSubnet1"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Remove-AzPostgreSqlFirewallRule.Tests.ps1:
```diff
@@ -22,7 +22,7 @@ Describe 'Remove-AzPostgreSqlFirewallRule' {
     It 'DeleteViaIdentity' {
         { 
             New-AzPostgreSqlFirewallRule -Name $env.firewallRuleName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -EndIPAddress 0.0.0.1 -StartIPAddress 0.0.0.0
-            $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/firewallRules/$($env.firewallRuleName)"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Remove-AzPostgreSqlServer.Tests.ps1:
```diff
@@ -26,7 +26,7 @@ Describe 'Remove-AzPostgreSqlServer' {
             #[SuppressMessage("Microsoft.Security", "CS002:SecretInNextLine")]
             $password = 'Pa88word!' | ConvertTo-SecureString -AsPlainText -Force
             New-AzPostgreSqlServer -Name postgresqldelete -ResourceGroupName $env.resourceGroup -Location $env.location -AdministratorUserName pwsh -AdministratorLoginPassword $password -Sku $env.Sku
-            $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/postgresqldelete"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Remove-AzPostgreSqlVirtualNetworkRule.Tests.ps1:
```diff
@@ -13,17 +13,17 @@ while(-not $mockingPath) {
 
 Describe 'Remove-AzPostgreSqlVirtualNetworkRule' {
     It 'Delete' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/MySqlVNet/subnets/MysqlSubnet1"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Restart-AzPostgreSqlServer.Tests.ps1:
```diff
@@ -20,7 +20,7 @@ Describe 'Restart-AzPostgreSqlServer' {
 
     It 'RestartViaIdentity' {
         { 
-            $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/restart"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Update-AzPostgreSqlConfiguration.Tests.ps1:
```diff
@@ -18,7 +18,7 @@ Describe 'Update-AzPostgreSqlConfiguration' {
     }
 
     It 'UpdateViaIdentityExpanded' {
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/configurations/deadlock_timeout"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Update-AzPostgreSqlFirewallRule.Tests.ps1:
```diff
@@ -22,7 +22,7 @@ Describe 'Update-AzPostgreSqlFirewallRule' {
 
     It 'UpdateViaIdentityExpanded' {
         New-AzPostgreSqlFirewallRule -Name $env.firewallRuleName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -EndIPAddress 0.0.0.1 -StartIPAddress 0.0.0.0
-        $ID = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/firewallRules/$($env.firewallRuleName)"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/test/Update-AzPostgreSqlVirtualNetworkRule.Tests.ps1:
```diff
@@ -13,19 +13,19 @@ while(-not $mockingPath) {
+        $SubnetID1 = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/PostgreSqlVnet/subnets/MysqlSubnet1"
         New-AzPostgreSqlVirtualNetworkRule -Name $env.VNetName -ResourceGroupName $env.resourceGroup -ServerName $env.serverName -SubnetId $SubnetID1
-        $SubnetID2 = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.Network/virtualNetworks/MySqlVNet/subnets/MysqlSubnet2"
-        $VNetId = "/subscriptions/$($env.SubscriptionId)/resourceGroups/$($env.resourceGroup)/providers/Microsoft.DBforPostgreSQL/servers/$($env.serverName)/virtualNetworkRules/$($env.VNetName)"
```
Revert manual change of the id. You can skip the similar case currently.

msJinLei commented on src/PostgreSql/custom/New-AzPostgreSqlFlexibleServer.ps1:
```diff
@@ -0,0 +1,596 @@
+            if (![string]::IsNullOrEmpty($FirewallRuleName)) {
+                $Server.FirewallRuleName = $FirewallRuleName
+            }
+            $Server.SecuredPassword =  $PSBoundParameters.AdministratorLoginPassword | ConvertTo-SecureString -AsPlainText -Force
```
No Database name needed here?

DaeunYim replied:
No database CRUD for PG now. It;s only recently implemented, that it is not included in the current swagger file

DaeunYim commented on src/PostgreSql/docs/Restore-AzPostgreSqlFlexibleServer.md:
````diff
@@ -0,0 +1,220 @@
+## SYNTAX
+
+```
+Restore-AzPostgreSqlFlexibleServer -Name <String> -ResourceGroupName <String> -SourceServerName <String>
````
Yes, this  is from backend difference.. PG only accepts server name while MySQL also accepts server Id.

msJinLei reviewed (changes requested):
Please generate use  autorest.cmd --use:@autorest/powershell@3.0.414. We have just upgraded the autorest.powershell

DaeunYim commented on src/PostgreSql/utils/adjectives.txt:
```diff
@@ -0,0 +1,166 @@
+aboard
```
Will do!

DaeunYim commented:
> Please generate use autorest.cmd --use:@autorest/powershell@3.0.414. We have just upgraded the autorest.powershell

Rebuilt and pushed a change with the txt files
